### PR TITLE
Add grpc+gdr protocol

### PR DESCRIPTION
**Pull request by CheukNgai**
I added grpc+gdr into validation set, which selects gdr as networking protocol. You can see more in [GDR pull request](https://github.com/tensorflow/tensorflow/pull/11392)

### CODE REVIEW

File: tensorflow_estimator/python/estimator/run_config.py [hunk 1]
```diff
@@ -293,7 +293,7 @@ def _validate(property_name, cond, message):
                     ' one argument "op".')
 
   _validate('protocol',
-            lambda protocol: protocol in (None, "grpc", "grpc+verbs"),
+            lambda protocol: protocol in (None, "grpc", "grpc+gdr", "grpc+verbs"),
             message='protocol should be grpc or grpc+verbs')
```
Comment (karmel) [inline tensorflow_estimator/python/estimator/run_config.py]: Update the message too?

**Comment by karmel** [comment]
This looks reasonable to me-- @poxvoculi , I presume this is okay from the perspective of the downstream grpc+gdr protocol?

File: tensorflow_estimator/python/estimator/run_config.py (hunk 1 above)
Comment (poxvoculi) [reply tensorflow_estimator/python/estimator/run_config.py]: @karmel  This seems fine to me.

**Comment by byronyi** [comment]
Do we have to validate this field in Python? It seems redundant as we already have error prompt in C++. Even the validated ones may not be usable if core TF is not compiled with that protocol support.

As a side note, we have many protocols besides grpc, enabled by plugins in tensorflow/networking. Considering tf-estimator resides in its own repo now, it is more difficult to sync for the maintainers.

File: tensorflow_estimator/python/estimator/run_config.py (hunk 1 above)
Comment (CheukNgai) [reply tensorflow_estimator/python/estimator/run_config.py]: @byronyi byron, would you pls follow this issue for me? Because "grpc+gdr" is implemented by you, it's better for you to discuss the design of upper layer caller API. Thanks, byron!!! :-)

**Comment by karmel** [comment]
@byronyi That is fair-- can you clarify what error gets raised from the C++ side, and when? Part of the goal of raising errors from the python side is to catch issues early and message clearly, rather than only deep down the stack after some amount of training time.

**Comment by byronyi** [comment]
@karmel You could find the error message [here](https://github.com/tensorflow/tensorflow/blob/master/tensorflow/core/distributed_runtime/server_lib.cc#L64) and the corresponding test [there](https://github.com/tensorflow/tensorflow/blob/master/tensorflow/core/distributed_runtime/server_lib_test.cc#L43).

I could submit a PR against it if it feels unclear though.

**Comment by karmel** [comment]
Thanks, @byronyi . As much as I hate the duplication, though, from the user perspective, the low-level errors raised from the C++ side are going to be hard to act on. Users setting up the run config won't necessarily now where that came from, I presume, whereas the error message here at least points out what in the run config needs to be changed. Is there any way to pull the set of options directly, as is done on the C++ side, so that we can point to where the error is, but not manually recreate the list?

**Comment by byronyi** [comment]
> Thanks, @byronyi . As much as I hate the duplication, though, from the user perspective, the low-level errors raised from the C++ side are going to be hard to act on. Users setting up the run config won't necessarily now where that came from, I presume, whereas the error message here at least points out what in the run config needs to be changed. Is there any way to pull the set of options directly, as is done on the C++ side, so that we can point to where the error is, but not manually recreate the list?

Thanks for your reply. I suppose we could expose a list of available server factories through TF C API. AFAIK @annarev is working on this; she might have a better idea.

**Comment by rthadur** [comment]
@CheukNgai is this PR still valid ?

**Comment by rthadur** [comment]
I'm going to go ahead and close this PR, because it seems to have stalled. If you're still interested in pursing this (and responding to my comments), please feel free to reopen!
